### ISSUE TITLE
Data caching without in-memory storage

### ISSUE THREAD

**Issue by EW0JY** [cabrerahector/wordpress-popular-posts#213]
Hi!
Under "Data Caching" I've had "Enable Caching" and "Refresh cache every: 24 hours" selected for a long time now. As far as I can tell, there were no issues.

However, just now I've come across the performance section regarding Caching, where it says:

> This feature is considered Beta as it hasn't been fully tested yet.
To enable the pageviews cache:
>1. you need access to an in-memory data storage, like Redis or Memcached (check with your hosting provider);
>2. install and set up a persistent object cache plugin on your site (eg. Redis Object Cache by Till Krüss, or Memcached Object Cache by Automattic);
>3. add define('WPP_CACHE_VIEWS', true); to your site's wp-config.php file anywhere before the /* That's all, stop editing! Happy blogging. */ message.

>After enabling the feature, WordPress Popular Posts will start storing your views data in-memory immediately. Setting WPP_CACHE_VIEWS to false (or removing the constant entirely from your wp-config.php file) will disable the feature.

I had not read that before and therefore have not done any of the steps mentioned.

Does my not following any of the steps mean that the enabled caching option actually doesn't do anything?

Thank you!

**Comment by cabrerahector** [comment]
Hi there!

Those instructions are related to the [Pageviews Cache feature](https://github.com/cabrerahector/wordpress-popular-posts/wiki/7.-Performance#caching-pageviews) which has nothing to do with the [Data Cache feature](https://github.com/cabrerahector/wordpress-popular-posts/wiki/7.-Performance#caching-db-queries-results), they're two different things:

- the **Pageviews Cache** stores page views data in-memory to reduce the number of database write operations, 
- the **Data Cache** stores database query results in cache to reduce the number of database read operations.

The Pageviews Cache is a new feature, it was introduced with WordPress Popular Posts 4.2.0 which is why you didn't see those instructions before.

**Comment by EW0JY** [comment]
Hi Hector,

sorry for the confusion, I got the distinction now.

Is there any way to make the Data Cache work with a CDN?
Specifically, I'm using KeyCDN and like I said, I've set the Data Cache to 24 hours.

However, even though most of the site is already in cache on the CDN servers, there's always an fresh request to the origin server from the Popular Posts plugin.
I can see that from waterfall diagrams (like gtmetrix's) and this request by WPP adds significantly to the page load time.

Is it possible to have the WPP cached data stored on the CDN?

Thanks so much!
